app/renderer/actions/Session.js

```diff
@@ -724,6 +725,32 @@ export function setSavedServerParams () {
   };
 }
 
+export function setStateFromAppiumJson (appiumJson) {
```

Comment:
if these significantly duplicate logic or functionality from setting the state via URL, perhaps they could be merged.